content-src/lib/selectLayoutRender.js

```diff
@@ -152,12 +184,11 @@ export const selectLayoutRender = (state, prefs, rickRollCache) => {
       for (const component of row.components.filter(
         c => !filterArray.includes(c.type)
       )) {
-        if (component.feed) {
-          const spocsConfig = component.spocs;
-          // Are we still waiting on a feed/spocs, render what we have,
-          // add a placeholder for this component, and bail out early.
+        const spocsConfig = component.spocs;
+        if (spocsConfig || component.feed) {
+          // TODO make sure this still works for different loading cases.
           if (
-            !feeds.data[component.feed.url] ||
+            (component.feed && !feeds.data[component.feed.url]) ||
             (spocsConfig &&
               spocsConfig.positions &&
               spocsConfig.positions.length &&
```

Comment:
This assumes layout will have spocs.positions defined for all components irrespective of their need for it e.g. TextPromo and TopSites. Had to change layout to this to make data passed to "top-sites" work https://gist.githubusercontent.com/punamdahiya/0de1900943a03e7acc17d9b766c76dc2/raw/3ab9355307baccfa3755168d5cb99fdff17908ba/stage-placement-index.json